core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java

```diff
@@ -1416,15 +1416,22 @@ public boolean addAll(Collection<? extends Entry<K, V>> c) {
 
       @Override
       public CacheStream<Entry<K, V>> stream() {
-         return new LocalEntryCacheStream(SimpleCacheImpl.this, false, null, getStreamSupplier(false), getComponentRegistry());
+         return cacheStreamCast(new LocalCacheStream<>(new EntryStreamSupplier<>(SimpleCacheImpl.this, null,
+                 getStreamSupplier(false)), false, componentRegistry));
       }
 
       @Override
       public CacheStream<Entry<K, V>> parallelStream() {
-         return new LocalEntryCacheStream(SimpleCacheImpl.this, true, null, getStreamSupplier(true), getComponentRegistry());
+         return cacheStreamCast(new LocalCacheStream<>(new EntryStreamSupplier<>(SimpleCacheImpl.this, null,
+                 getStreamSupplier(false)), true, componentRegistry));
       }
    }
 
+   // This is a hack to allow the cast to work.  Java doesn't like subtypes in generics
```

Comment:
I can't define the return type as that though :/  The problem is that return type for Map wants Entry which we can't change.